src/main/java/ch/njol/skript/effects/EffToggle.java

```diff
@@ -71,15 +78,17 @@
 		setDataMethod = mh;
 	}
 	
-	@SuppressWarnings("null")
-	private Expression<Block> blocks;
+	private Expression<?> toggledExpr;
 	private int toggle;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		blocks = (Expression<Block>) vars[0];
+		toggledExpr = (Expression<?>) vars[0];
 		toggle = matchedPattern - 1;
+		if (toggledExpr.getReturnType() == Boolean.class && !CollectionUtils.containsSuperclass(toggledExpr.acceptChange(ChangeMode.SET), Boolean.class)) {
```

Comment:
Return type of toggledExpr isn't guaranteed Boolean when it'll be executed with booleans, e.g. if it's a variable or any other expression that is more flexible with its return type. I recommend you check with `toggledExpr.getReturnType().isAssignableFrom(Boolean.class)`.